src/app/features/tech-record/components/tech-record-amend-vrm/tech-record-amend-vrm.component.ts

```diff
@@ -85,34 +90,48 @@ export class AmendVrmComponent implements OnInit, OnChanges {
     this.router.navigate(['..'], { relativeTo: this.route });
   }
 
-  handleSubmit(newVrm: string): void {
+  handleSubmit(newVrm: string, cherishedTransfer: string): void {
+    console.log(cherishedTransfer);
     this.globalErrorService.clearErrors();
     if (newVrm === '' || (newVrm === this.vrm ?? '')) {
-      return this.globalErrorService.addError({ error: 'You must provide a new VRM', anchorLink: 'newVrm' });
+      this.globalErrorService.addError({ error: 'You must provide a new VRM', anchorLink: 'newVrm' });
+    }
+    if (cherishedTransfer === '' || cherishedTransfer === undefined) {
+      this.globalErrorService.addError({ error: 'You must provide a reason for amending', anchorLink: 'cherishedTransfer' });
     }
 
-    this.technicalRecordService
-      .isUnique(newVrm, SEARCH_TYPES.VRM)
-      .pipe(
-        take(1),
-        catchError(error => (error.status == 404 ? of(true) : throwError(() => new Error('Error'))))
-      )
-      .subscribe({
-        next: res => {
-          if (!res) return this.globalErrorService.addError({ error: 'VRM already exists', anchorLink: 'newVrm' });
-
-          const newVehicleRecord = this.amendVrm(this.vehicle!, newVrm);
-
-          this.setReasonForCreation(newVehicleRecord);
-          this.technicalRecordService.updateEditingTechRecord({ ...newVehicleRecord });
-          this.store.dispatch(updateTechRecords({ systemNumber: this.vehicle!.systemNumber }));
-        },
-        error: e => this.globalErrorService.addError({ error: 'Internal Server Error', anchorLink: 'newVrm' })
-      });
+    this.globalErrorService.errors$.pipe(take(1)).subscribe({
+      next: res => {
+        if (res.length > 0) return;
+        this.technicalRecordService
+          .isUnique(newVrm, SEARCH_TYPES.VRM)
+          .pipe(
+            take(1),
+            catchError(error => (error.status == 404 ? of(true) : throwError(() => new Error('Error'))))
+          )
+          .subscribe({
+            next: res => {
+              if (!res) return this.globalErrorService.addError({ error: 'VRM already exists', anchorLink: 'newVrm' });
+
+              const newVehicleRecord = this.amendVrm(this.vehicle!, newVrm, cherishedTransfer === 'true');
```

Comment:
Why do we only report an error if cherishedTransfer is empty without returning early? Are we trying to run this code regardless of the errors?